var/spack/repos/builtin/packages/bigdft-atlab/package.py

```diff
@@ -39,20 +42,31 @@ def configure_args(self):
         prefix = self.prefix
 
         fcflags = []
+        cflags = []
+        cxxflags = []
+
         if "+openmp" in spec:
             fcflags.append(self.compiler.openmp_flag)
 
+        if "+shared" in spec:
+            fcflags.append("-fPIC")
+            cflags.append("-fPIC")
+            cxxflags.append("-fPIC")
         if self.spec.satisfies("%gcc@10:"):
             fcflags.append("-fallow-argument-mismatch")
 
         args = [
             "FCFLAGS=%s" % " ".join(fcflags),
+            "CFLAGS=%s" % " ".join(cflags),
+            "CXXFLAGS=%s" % " ".join(cxxflags),
```

Comment:
FYI.  We are recommending use of python f-strings for formatting as we convert the spack core to use them.  See https://docs.python.org/3/tutorial/inputoutput.html.